### PR TITLE
AMD/Xilinx Versal ACAP: ECC

### PR DESCRIPTION
**Pull request by ldts**

### CODE REVIEW

**Comment by ldts** [comment]
code style is a false positive

**Comment by nathan-menhorn** [comment]
Hi @jbech-linaro any updates on the PR? Thanks.

**Review by jenswi-linaro — COMMENTED** [review]
The commit "libtomcrypt/mbedtls: ecc: support the crypto driver" have to be split into three commits (one for the .h file, one for LTC and one for mbedtls) since the LTC and mbedtls changes must be cherry-picked into separate import branches when LTC or mbdeltls are updated.

File: core/drivers/crypto/versal/ecc.c [hunk 1]
```diff
@@ -0,0 +1,450 @@
...
+					    sizeof(buf));
+		if (ret)
+			return ret;
+		break;
+	default:
+		return TEE_ERROR_GENERIC;
+	}
+
+	/* Write the hash - or msg - to the message buffer */
+	versal_mbox_alloc(len, msg_len == len ? msg : buf, p);
```
[the remark this answers is not in the document]
Comment (ldts) [reply core/drivers/crypto/versal/ecc.c]: I agree, it is more clear without it.

**Comment by ldts** [comment]
anything else?

File: core/include/crypto/crypto_impl.h [hunk 2]
```diff
@@ -355,6 +355,12 @@ struct crypto_ecc_keypair_ops {
 };
 
 #ifdef CFG_CRYPTO_ECC
+const struct crypto_ecc_keypair_ops *
+crypto_asym_get_ecc_keypair_ops(uint32_t key_type);
+
+const struct crypto_ecc_public_ops*
```
Comment (jenswi-linaro) [inline core/include/crypto/crypto_impl.h:361]: Please add a space before the `*`, same below.

**Review by jenswi-linaro — COMMENTED** [review]
With comment addressed please apply:
`Acked-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Pull request merged**